### PR TITLE
mpi: rename mpi 4.0 functions from MPIX to MPI prefix

### PR DESCRIPTION
**Pull request by hzhou**
## Pull Request Description

The main branch is targeting for the mpich 4.0 release, and it assumes MPI 4.0 support. Thus we don't need MPIX_ prefix for the new functions.

## Author Checklist
* [x] Reference appropriate issues (with "Fixes" or "See" as appropriate)
* [x] Remove xfail from the test suite when fixing a test
* [x] Commits are self-contained and do not do two things at once
* [x] Commit message is of the form: `module: short description` and follows [good practice](https://chris.beams.io/posts/git-commit/)
* [ ] Passes whitespace checkers
* [ ] Passes warning tests
* [ ] Passes all tests
* [x] Add comments such that someone without knowledge of the code could understand
* [x] You or your company has a signed contributor's agreement on file with Argonne
* [x] For non-Argonne authors, request an explicit comment from your companies PR approval manager

### CODE REVIEW

**Comment by hzhou** [comment]
test:mpich/ch4/most
test:mpich/ch3/most

File: src/binding/fortran/use_mpi_f08/mpi_c_interface_nobuf.f90 [hunk 1]
```diff
@@ -1384,15 +1384,15 @@ function MPIR_Add_error_string_c(errorcode, string) &
 end function MPIR_Add_error_string_c
 
 function MPIR_Delete_error_class_c(errorclass) &
-    bind(C, name="PMPIX_Delete_error_class") result(ierror)
+    bind(C, name="PMPI_Delete_error_class") result(ierror)
```
Comment (raffenet) [inline src/binding/fortran/use_mpi_f08/mpi_c_interface_nobuf.f90:1387]: These delete functions are not in MPI-4.0. That should retain the X for now.

Comment (hzhou) [reply src/binding/fortran/use_mpi_f08/mpi_c_interface_nobuf.f90:1387]: Okay

Comment (hzhou) [reply src/binding/fortran/use_mpi_f08/mpi_c_interface_nobuf.f90:1387]: In this case, I'll close this PR and re-introduce the comm commits in the split comm api autogen PR.

**Pull request closed without merging**